src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -114,37 +114,87 @@ public class ChangeManagement implements Serializable {
             return returnValue;
 
         }
-
     }
 
     void releaseTransportRequest(String changeId, String transportRequestId, String endpoint, String credentialsId, String clientOpts = '') {
-        int rc = executeWithCredentials( endpoint, credentialsId, 'release-transport', ['-cID', changeId,
-                                                                                        '-tID', transportRequestId], clientOpts) as int
+        int rc = executeWithCredentials( endpoint, credentialsId, 'release-transport', [new KeyValue('cID', changeId),
+                                                                                        new KeyValue('tID', transportRequestId)], clientOpts) as int
         if(rc == 0) {
             return
         } else {
             throw new ChangeManagementException("Cannot release Transport Request '$transportRequestId'. Return code from cmclient: $rc.")
         }
     }
-
+    
     String getCMCommandLine(String endpoint,
                             String username,
                             String password,
                             String command,
-                            List<String> args,
+                            List<CLOption> args,
                             String clientOpts = '') {
         String cmCommandLine = '#!/bin/bash'
         if(clientOpts) {
             cmCommandLine +=  """
                              export CMCLIENT_OPTS="${clientOpts}" """
         }
+        
         cmCommandLine += """
                         cmclient -e '$endpoint' \
                            -u '$username' \
                            -p '$password' \
                            -t SOLMAN \
-                          ${command} ${(args as Iterable).join(' ')}
+                          ${command} ${args.collect{it}.join(' ')}
                     """
+        
         return cmCommandLine
     }
+    
+    abstract static class CLOption 
+    {
+        String key=null
+        String value=null
+        boolean quoteValue=true;
+        String hyphen='-'
```

Comment:
Since we do not have other hyphens (convince me from the opposite) I vote for making this static final.  The term hyphen is bogus since a hyphen is a ```-``` and setting it to e.g. ```abc``` is hard to explain.

Reply:
removed